src/main/java/be/seeseemelk/mockbukkit/block/banner/PatternTypeMock.java

```diff
@@ -0,0 +1,46 @@
+package be.seeseemelk.mockbukkit.block.banner;
+
+import be.seeseemelk.mockbukkit.util.OldKeyedEnumMock;
+import com.google.gson.JsonObject;
+import org.bukkit.NamespacedKey;
+import org.bukkit.block.banner.PatternType;
+import org.jetbrains.annotations.NotNull;
+
+public class PatternTypeMock extends OldKeyedEnumMock<PatternType> implements PatternType
+{
+
+	private final NamespacedKey key;
+	private final String identifier;
+
+	public PatternTypeMock(NamespacedKey key,String name, int ordinal,String identifier)
+	{
+		super(name, ordinal,key);
+		this.identifier = identifier;
+		this.key = key;
+	}
+
```

Comment:
Probably best to add the @ApiStatus.Internal annotation to this // from my point of view this method should not be used in any way except internally